DataTests/FavoriteTests.swift

```diff
@@ -0,0 +1,227 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+import XCTest
+@testable import Data
+
+class FavoriteTests: CoreDataTestCase {
+    
+    private let fetchRequest: NSFetchRequest<Bookmark> = {
+        let fetchRequest = NSFetchRequest<Bookmark>(entityName: String(describing: Bookmark.self))
+        // We always want favorites folder to be on top, in the first section.
+        fetchRequest.sortDescriptors = [
+            NSSortDescriptor(key: "order", ascending: true),
+            NSSortDescriptor(key: "created", ascending: false)
+        ]
+        fetchRequest.predicate = NSPredicate(format: "isFavorite == true")
+        return fetchRequest
+    }()
+    
+    private lazy var fetchController = NSFetchedResultsController<Bookmark>(
```

Comment:
Nice job wrapping those long lines, reviewing PRs without horizontal scrolling is always nice